### PR TITLE
[feature](iceberg) support iceberg equality delete

### PR DESCRIPTION
**Pull request by AshinGau**
## Proposed changes

Support iceberg equality delete.
If there's only one delete column in delete file, use `SimpleEqualityDelete`, which uses optimized `HybridSetBase` to build the hash set. If there are more delete columns in delete file, use `MultiEqualityDelete`, which generates a hash column from all delete columns, and only compare the values when the hash values are the same.

### Remaining Issues
1. Support type changes in delete columns(todo)
2. Support column rename in delete columns(todo)
3. Test large data files with large delete files(todo)

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Comment by AshinGau** [comment]
run buildall

**Review by morningman — DISMISSED** [review]
LGTM

**Review by zddr — APPROVED** [review]

File: be/src/vec/exec/format/table/equality_delete.cpp [hunk 1]
```diff
@@ -0,0 +1,156 @@
...
+Status MultiEqualityDelete::_build_set() {
+    COUNTER_UPDATE(num_delete_rows, _delete_block->rows());
+    size_t rows = _delete_block->rows();
+    _delete_hashes.clear();
+    _delete_hashes.resize(rows, 0);
+    for (ColumnPtr column : _delete_block->get_columns()) {
+        column->update_hashes_with_value(_delete_hashes.data(), nullptr);
+    }
+    for (size_t i = 0; i < rows; ++i) {
+        _delete_hash_map[_delete_hashes[i]] = i;
```
Comment (BiteTheDDDDt) [inline be/src/vec/exec/format/table/equality_delete.cpp:98]: what if hash collision

**Review by wuwenchi — APPROVED** [review]

**Comment by AshinGau** [comment]
run buildall

File: be/src/vec/exec/format/table/equality_delete.cpp (hunk 1 above)
Comment (AshinGau) [reply be/src/vec/exec/format/table/equality_delete.cpp]: use `std::multimap` instead.

**Review by BiteTheDDDDt — APPROVED** [review]

**Pull request merged**